GrpcClientConfiguration.inProcess results in NPE when config not present

### Describe the bug

If `quarkus-grpc-inprocess` is on the classpath, you have a grpc test that needs to test with real ports and you have not explicitly declared the property to disable inprocess, e.g. 

```
quarkus.grpc.clients.foo.in-process.enabled=false
```

...then they fail with:

```
Caused by: java.lang.NullPointerException: Cannot read field "enabled" because "configuration.inProcess" is null
	at io.quarkus.grpc.inprocess.InProcessGrpcServerBuilderProvider.providesChannel(InProcessGrpcServerBuilderProvider.java:72)
```

I would expect it to honour the default value of enabled (false) if the config is not present.

Should `GrpcClientConfiguration.inProcess` be `Optional` or should `InProcessGrpcServerBuilderProvider.providesChannel` defend against the NPE? Or is there a way to express the `GrpcClientConfiguration.inProcess` `@ConfigItem` whereby `InProcess` will be a valid default instance?

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/grpc/inprocess/src/main/java/io/quarkus/grpc/inprocess/InProcessGrpcServerBuilderProvider.java
+++ b/extensions/grpc/inprocess/src/main/java/io/quarkus/grpc/inprocess/InProcessGrpcServerBuilderProvider.java
@@ -12,6 +12,7 @@ import io.grpc.ServerMethodDefinition;
 import io.grpc.ServerServiceDefinition;
 import io.grpc.inprocess.InProcessChannelBuilder;
 import io.grpc.inprocess.InProcessServerBuilder;
+import io.quarkus.grpc.runtime.config.Enabled;
 import io.quarkus.grpc.runtime.config.GrpcClientConfiguration;
 import io.quarkus.grpc.runtime.config.GrpcServerConfiguration;
 import io.quarkus.grpc.spi.GrpcBuilderProvider;
@@ -24,7 +25,7 @@ import io.vertx.core.impl.VertxInternal;
 public class InProcessGrpcServerBuilderProvider implements GrpcBuilderProvider<InProcessServerBuilder> {
     @Override
     public boolean providesServer(GrpcServerConfiguration configuration) {
-        return configuration.inProcess.enabled;
+        return Enabled.isEnabled(configuration.inProcess);
     }
 
     @Override
@@ -69,7 +70,7 @@ public class InProcessGrpcServerBuilderProvider implements GrpcBuilderProvider<I
 
     @Override
     public boolean providesChannel(GrpcClientConfiguration configuration) {
-        return configuration.inProcess.enabled;
+        return Enabled.isEnabled(configuration.inProcess);
     }
 
     @Override

--- a/extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/Enabled.java
+++ b/extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/Enabled.java
@@ -1,0 +1,9 @@
+package io.quarkus.grpc.runtime.config;
+
+public interface Enabled {
+    boolean isEnabled();
+
+    static boolean isEnabled(Enabled enabled) {
+        return enabled != null && enabled.isEnabled();
+    }
+}

--- a/extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/InProcess.java
+++ b/extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/InProcess.java
@@ -2,13 +2,21 @@ package io.quarkus.grpc.runtime.config;
 
 import io.quarkus.runtime.annotations.ConfigGroup;
 import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.IgnoreProperty;
 
 /**
  * In-process config
  * * <a href="https://grpc.github.io/grpc-java/javadoc/io/grpc/inprocess/InProcessServerBuilder.html">in-process usage</a>
  */
 @ConfigGroup
-public class InProcess {
+public class InProcess implements Enabled {
+
+    @Override
+    @IgnoreProperty
+    public boolean isEnabled() {
+        return enabled;
+    }
+
     /**
      * Explicitly enable use of in-process.
      */

--- a/extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/Xds.java
+++ b/extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/Xds.java
@@ -2,13 +2,21 @@ package io.quarkus.grpc.runtime.config;
 
 import io.quarkus.runtime.annotations.ConfigGroup;
 import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.IgnoreProperty;
 
 /**
  * XDS config
  * * <a href="https://github.com/grpc/grpc-java/tree/master/examples/example-xds">XDS usage</a>
  */
 @ConfigGroup
-public class Xds {
+public class Xds implements Enabled {
+
+    @Override
+    @IgnoreProperty
+    public boolean isEnabled() {
+        return enabled;
+    }
+
     /**
      * Explicitly enable use of XDS.
      */

--- a/extensions/grpc/xds/src/main/java/io/quarkus/grpc/xds/XdsGrpcServerBuilderProvider.java
+++ b/extensions/grpc/xds/src/main/java/io/quarkus/grpc/xds/XdsGrpcServerBuilderProvider.java
@@ -21,6 +21,7 @@ import io.grpc.xds.XdsChannelCredentials;
 import io.grpc.xds.XdsServerBuilder;
 import io.grpc.xds.XdsServerCredentials;
 import io.quarkus.grpc.runtime.config.ClientXds;
+import io.quarkus.grpc.runtime.config.Enabled;
 import io.quarkus.grpc.runtime.config.GrpcClientConfiguration;
 import io.quarkus.grpc.runtime.config.GrpcServerConfiguration;
 import io.quarkus.grpc.runtime.config.Xds;
@@ -37,8 +38,7 @@ import io.vertx.core.impl.VertxInternal;
 public class XdsGrpcServerBuilderProvider implements GrpcBuilderProvider<XdsServerBuilder> {
     @Override
     public boolean providesServer(GrpcServerConfiguration configuration) {
-        Xds xds = configuration.xds;
-        return xds != null && xds.enabled;
+        return Enabled.isEnabled(configuration.xds);
     }
 
     @Override
@@ -96,12 +96,7 @@ public class XdsGrpcServerBuilderProvider implements GrpcBuilderProvider<XdsServ
 
     @Override
     public boolean providesChannel(GrpcClientConfiguration configuration) {
-        Xds xds = configuration.xds;
-        if (xds != null) {
-            return xds.enabled || XDS.equalsIgnoreCase(configuration.nameResolver);
-        } else {
-            return false;
-        }
+        return Enabled.isEnabled(configuration.xds) || XDS.equalsIgnoreCase(configuration.nameResolver);
     }
 
     @Override

